airflow/utils/dag_processing.py

```diff
@@ -1309,3 +1315,81 @@ def end(self):
                     self.log.info("Killing child PID: %s", child.pid)
                     child.kill()
                     child.wait()
+
+
+def process_dag_file(filepath, safe_mode=True):
```

Comment:
Personally, I'm not a fan of separate functions, can't we merge this into a class as a static method? For example the `DagFileProcessorAgent`.

Reply:
For functionality this does not matter. Adding a class that will never be initialised feels wrong for me. If this is part of a class this would make sense but this is an independent method that can be used in multiple location and not bound to a single class.

Still if this is a blocking issue I can move to an existing class or make a new class for this.